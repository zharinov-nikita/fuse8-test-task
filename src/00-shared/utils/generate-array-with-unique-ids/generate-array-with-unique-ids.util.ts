export function utilGenerateArrayWithUniqueIds(length: number) {
    return Array.from({ length }, (_, index) => ({ id: index }))
}
