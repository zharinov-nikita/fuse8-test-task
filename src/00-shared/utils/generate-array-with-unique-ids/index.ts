export * from './generate-array-with-unique-ids.util'
