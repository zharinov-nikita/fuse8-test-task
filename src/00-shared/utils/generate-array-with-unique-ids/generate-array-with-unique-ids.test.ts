import { describe, test, expect } from 'vitest'
import { sharedUtilGenerateArrayWithUniqueIds } from './generate-array-with-unique-ids.util'

describe('sharedUtilFormatDate', () => {
    const data = 10
    const result = [
        { id: 0 },
        { id: 1 },
        { id: 2 },
        { id: 3 },
        { id: 4 },
        { id: 5 },
        { id: 6 },
        { id: 7 },
        { id: 8 },
        { id: 9 },
    ]
    test('case 1', () => {
        expect(sharedUtilGenerateArrayWithUniqueIds(data)).toEqual(result)
        expect(sharedUtilGenerateArrayWithUniqueIds(data)).toMatchSnapshot()
    })
})
