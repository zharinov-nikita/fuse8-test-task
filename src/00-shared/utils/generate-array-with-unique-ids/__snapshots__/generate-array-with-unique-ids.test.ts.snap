// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sharedUtilFormatDate > case 1 1`] = `
[
  {
    "id": 0,
  },
  {
    "id": 1,
  },
  {
    "id": 2,
  },
  {
    "id": 3,
  },
  {
    "id": 4,
  },
  {
    "id": 5,
  },
  {
    "id": 6,
  },
  {
    "id": 7,
  },
  {
    "id": 8,
  },
  {
    "id": 9,
  },
]
`;

exports[`sharedUtilFormatDate > snapshot 1`] = `
[
  {
    "id": 0,
  },
  {
    "id": 1,
  },
  {
    "id": 2,
  },
  {
    "id": 3,
  },
  {
    "id": 4,
  },
  {
    "id": 5,
  },
  {
    "id": 6,
  },
  {
    "id": 7,
  },
  {
    "id": 8,
  },
  {
    "id": 9,
  },
]
`;
